README.md

```diff
@@ -149,9 +149,9 @@ These methods are available on a Writer object:
   Connect to the nsqd specified.
 * `close()` <br/>
   Disconnect from nsqd.
-* `publish(topic, msgs, [callback])` <br/>
+* `publish(topic, msgs, [timeMs], [callback])` <br/>
```

Comment:
I'm also leaning into that direction from an API standpoint. But from a maintainabilty I lean more towards this solution. Because you have to duplicate multiple checks for validity.

Another option: create `_publish` which gets called by `deferPublish` and `publish`. And each of the public methods is more or less direcly calling `_publish` with the correct params.